clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ListOffsetsEvent.java

```diff
@@ -32,8 +32,7 @@
  * {@link OffsetAndTimestamp} found (offset of the first message whose timestamp is greater than
  * or equals to the target timestamp)
  */
-public class ListOffsetsEvent extends CompletableApplicationEvent<Map<TopicPartition, OffsetAndTimestamp>> {
-
+public class ListOffsetsEvent<T> extends CompletableApplicationEvent<Map<TopicPartition, T>> {
```

Comment:
How about using two separate events `ListOffsets` and `ListOffsetsWithTimestamps`? We could save the "requireTimestamps" boolean and the use of generics here, which I think would simplify the code. That would also get rid of some `unchecked` warnings, that you are currently suppressing, I'd expect.